lib/server.js

```diff
@@ -108,19 +108,25 @@ exports.create = function create(createOptions) {
 
     try {
       if (this.useRouter) {
-        match(thing, function (error, redirectLocation, renderProps) {
+        match({ routes:createOptions.routes, location:thing},
+          function(error, redirectLocation, renderProps) {
           if (redirectLocation) {
+            debug('server.js match 302 %s', redirectLocation);
```

Comment:
@chunkiat82  do you have any suggestions on what can be done to address 302 and 404 ?